exporters/otlp/include/opentelemetry/exporters/otlp/protobuf_include_prefix.h

```diff
@@ -30,9 +33,32 @@
 #  endif
 #  pragma GCC diagnostic ignored "-Wunused-parameter"
 #  pragma GCC diagnostic ignored "-Wtype-limits"
+#  pragma GCC diagnostic ignored "-Wsign-compare"
+#  pragma GCC diagnostic ignored "-Wsign-conversion"
+#  pragma GCC diagnostic ignored "-Wshadow"
+#  pragma GCC diagnostic ignored "-Wuninitialized"
+#  pragma GCC diagnostic ignored "-Wconversion"
+#  if (__GNUC__ * 100 + __GNUC_MINOR__) >= 409
+#    pragma GCC diagnostic ignored "-Wfloat-conversion"
+#  endif
+#  if (__GNUC__ * 100 + __GNUC_MINOR__) >= 501
+#    pragma GCC diagnostic ignored "-Wsuggest-override"
+#  endif
 #elif defined(__clang__) || defined(__apple_build_version__)
 #  pragma clang diagnostic push
 #  pragma clang diagnostic ignored "-Wunused-parameter"
 #  pragma clang diagnostic ignored "-Wtype-limits"
 #  pragma clang diagnostic ignored "-Wshadow-field"
+#  pragma clang diagnostic ignored "-Wsign-compare"
+#  pragma clang diagnostic ignored "-Wsign-conversion"
+#  pragma clang diagnostic ignored "-Wshadow"
+#  pragma clang diagnostic ignored "-Wuninitialized"
+#  pragma clang diagnostic ignored "-Wconversion"
+#  if !(((__clang_major__ * 100) + __clang_minor__) >= 305)
+#    pragma GCC diagnostic ignored "-Wsuggest-override"
```

Comment:
Should this be `pragma clang` instead ?

Reply:
> Should this be pragma clang instead ? 

Thanks and fixed.

> is the intent to suppress the warning for versions before clang 3.5, and not after ?

No, these warnings are available after clang 3.5.

Reply:
@owent Thanks for the clarification.

Should
```
if !(((__clang_major__ * 100) + __clang_minor__) >= 305)
```

be changed to

```
if (((__clang_major__ * 100) + __clang_minor__) >= 305)
```

then, and likewise for 3.9  ?